lib/iris/fileformats/netcdf.py

```diff
@@ -1361,7 +1361,6 @@ def _add_inner_related_vars(
         cf_var_cube,
         dimension_names,
         coordlike_elements,
-        saver_create_method,
         role_attribute_name,
```

Comment:
@pp-mo We could take the opportunity to drop the `role_attribute_name` argument, and instead use the `element` within `_add_inner_related_vars` to lookup the `_ELEMENT_TYPE_NAMES`?

Reply:
:+1: I've now added code to test the type of the arguments